goal_src/jak1/pc/progress-pc.gc

```diff
@@ -1182,6 +1208,209 @@
+                  )
+                 (((game-text-id beach-level-name))
+                  (when (-> *pc-settings* speedrunner-mode?)
+                    (set! *speedrun-setup-function* (lambda ()
+                      ;; unlock blue eco vent
+                      (close-specific-task! (game-task jungle-eggtop) (task-status need-resolution))
+                      (send-event *target* 'get-pickup (pickup-type fuel-cell) (the float (game-task jungle-eggtop)))
+                      ;; extra setup for IL speedruns
+                      (speedrun-set-common-settings)
+                      ))
+                    )
+                  (initialize! *game-info* 'game (the-as game-save #f) "beach-start")
+                  )
+                 (((game-text-id jungle-level-name))
+                  (when (-> *pc-settings* speedrunner-mode?)
+                    (set! *speedrun-setup-function* (lambda ()
+                      ;; extra setup for IL speedruns
+                      (speedrun-set-common-settings)
+                      ))
+                    )
+                  (initialize! *game-info* 'game (the-as game-save #f) "jungle-start")
+                  )
+                 (((game-text-id misty-level-name))
+                  (when (-> *pc-settings* speedrunner-mode?)
+                    (set! *speedrun-setup-function* (lambda ()
+                      ;; unlock fisherman's boat
+                      (close-specific-task! (game-task jungle-fishgame) (task-status need-resolution))
+                      (send-event *target* 'get-pickup (pickup-type fuel-cell) (the float (game-task jungle-fishgame)))
+                      ;; extra setup for IL speedruns
+                      (speedrun-set-common-settings)
+                      ))
+                    )
+                  (initialize! *game-info* 'game (the-as game-save #f) "misty-start")
+                  )
+                 (((game-text-id firecanyon-level-name))
+                  (when (-> *pc-settings* speedrunner-mode?)
+                    (set! *speedrun-setup-function* (lambda ()
+                      ;; give 20 cells to unlock zoomer
+                      (set! (-> *game-info* fuel) 20.0)
+                      ;; skip keira intro cutscene
+                      (close-specific-task! (game-task firecanyon-assistant) (task-status need-reward-speech))
+                      ;; extra setup for IL speedruns
+                      (speedrun-set-common-settings)
+                      ))
+                    )
+                  (initialize! *game-info* 'game (the-as game-save #f) "firecanyon-start")
+                  )
+                 (((game-text-id village2-level-name))
+                  (when (-> *pc-settings* speedrunner-mode?)
+                    (set! *speedrun-setup-function* (lambda ()
+                      ;; give enough orbs to buy all cells 120+120+90+90+90=510
+                      (set! (-> *game-info* money-total) 510.0)
```

Comment:
Just a thought process but wouldn't ` (set! (-> *game-info* money-total) (+  (-> *game-info* money-total) 510.0))` be a safer practice here? It *Should* be zero, but what happens if it isn't?